Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindow.initializePanelMenu:COMMENT
Method Modifier: protected   internal    
Comment:/**
 * Initializes the menu associated with the given panel feature state. You
 * must at the very least set PanelFeatureState.menu to the Menu to be
 * associated with the given panel state. The default implementation creates
 * a new menu for the panel state.
 *
 * @param st The panel whose menu is being initialized.
 * @return Whether the initialization was successful.
 */

@@ -1,14 +1,30 @@
 {
     Context context = getContext();
-    // If we have an action bar, initialize the menu with a context themed for it.
-    if ((st.featureId == FEATURE_OPTIONS_PANEL || st.featureId == FEATURE_ACTION_BAR) && mActionBar != null) {
-        TypedValue outValue = new TypedValue();
-        Resources.Theme currentTheme = context.getTheme();
-        currentTheme.resolveAttribute(com.android.internal.R.attr.actionBarWidgetTheme, outValue, true);
-        final int targetThemeRes = outValue.resourceId;
-        if (targetThemeRes != 0 && context.getThemeResId() != targetThemeRes) {
-            context = new ContextThemeWrapper(context, targetThemeRes);
+    // If we have an action bar, initialize the menu with the right theme.
+    if ((st.featureId == FEATURE_OPTIONS_PANEL || st.featureId == FEATURE_ACTION_BAR) && mDecorContentParent != null) {
+        final TypedValue outValue = new TypedValue();
+        final Theme baseTheme = context.getTheme();
+        baseTheme.resolveAttribute(R.attr.actionBarTheme, outValue, true);
+        Theme widgetTheme = null;
+        if (outValue.resourceId != 0) {
+            widgetTheme = context.getResources().newTheme();
+            widgetTheme.setTo(baseTheme);
+            widgetTheme.applyStyle(outValue.resourceId, true);
+            widgetTheme.resolveAttribute(R.attr.actionBarWidgetTheme, outValue, true);
+        } else {
+            baseTheme.resolveAttribute(R.attr.actionBarWidgetTheme, outValue, true);
         }
+        if (outValue.resourceId != 0) {
+            if (widgetTheme == null) {
+                widgetTheme = context.getResources().newTheme();
+                widgetTheme.setTo(baseTheme);
+            }
+            widgetTheme.applyStyle(outValue.resourceId, true);
+        }
+        if (widgetTheme != null) {
+            context = new ContextThemeWrapper(context, 0);
+            context.getTheme().setTo(widgetTheme);
+        }
     }
     final MenuBuilder menu = new MenuBuilder(context);
     menu.setCallback(this);

